mllib/src/test/scala/org/apache/spark/ml/clustering/PowerIterationClusteringSuite.scala

```diff
@@ -169,6 +169,59 @@ class PowerIterationClusteringSuite extends SparkFunSuite
     assert(localAssignments === localAssignments2)
   }
 
+  test("Power Iteration failed to converge") {
+    /*
+         Graph:
+            1
+           /
+          /
+         0      2 -- 3
+     */
+    val data1 = spark.createDataFrame(Seq(
+      (0, 1),
+      (2, 3)
+    )).toDF("src", "dst")
+
+    val msg1 = intercept[SparkException] {
+      new PowerIterationClustering()
+        .setInitMode("random")
+        .setK(2)
+        .assignClusters(data1)
+    }.getMessage
+    assert(msg1.contains("Power Iteration fail to converge"))
```

Comment:
Is there anything else you can assert about the model to find that it didn't converge, rather than catch the exception? I agree we can't test for log messages. It could be fine to just assert that this produces some valid answer in a case like this, as a test, if it does not throw an exception.

Reply:
Thanks. I will try to write an alternative UT, in case of log.

Reply:
Yeah I'd log a warning and let it proceed. Anything you can assert about the model here is OK; it may not be possible to test much other than it succeeded

Reply:
Thanks. updated